translate/storage/php.py

```diff
@@ -216,171 +339,88 @@ def serialize(self, out):
-                                      self.filename, line_number,
-                                      keys_dict[location])
-                    else:
-                        keys_dict[location] = line_number
-
-                    # Add the location to the translation unit.
-                    newunit.addlocation(location)
-
-                    # Save the translation in the value variable.
-                    value = line[equalpos+len(equaldel):].lstrip()[1:]
-                    lastvalue = ""
-                    invalue = True
-            else:
-                # If no equalpos is present in the current line, but this is a
-                # multiline translation.
-                if invalue:
-                    value = line
-
-            # Get the end delimiter position.
-            enddelpos = value.rfind(enddel)
-
-            # Process the current line until all entries on it are parsed.
-            while enddelpos != -1:
-                # Check if the latest non-whitespace character before the end
-                # delimiter is the valuequote.
-                if value[:enddelpos].rstrip()[-1] == valuequote:
-                    # Save the value string without trailing whitespaces and
-                    # without the ending quotes.
-                    newunit.value = lastvalue + value[:enddelpos].rstrip()[:-1]
-                    newunit.escape_type = valuequote
-                    lastvalue = ""
-                    invalue = False
-
-                # If there is more text (a comment) after the translation.
-                if not invalue and enddelpos != (len(value) - 1):
-                    commentinlinepos = value.find("//", enddelpos)
-                    if commentinlinepos != -1:
-                        newunit.addnote(value[commentinlinepos+2:].strip(),
-                                        "developer")
-
-                # If the translation is already parsed, save it and initialize
-                # a new translation unit.
-                if not invalue:
-                    self.addunit(newunit)
-                    value = ""
-                    newunit = phpunit()
-
-                # Update end delimiter position to the previous last appearance
-                # of the end delimiter, because it might be several entries in
-                # the same line.
-                enddelpos = value.rfind(enddel, 0, enddelpos)
-            else:
-                # After processing current line, if we are not in an array,
-                # fall back to default dialect (PHP simple variable syntax).
-                if not inarray:
-                    equaldel = "="
-                    enddel = ";"
-
-            # If this is part of a multiline translation, just append it to the
-            # previous translation lines.
-            if invalue:
-                lastvalue = lastvalue + value + "\n"
+        def newunit(name, value, escape_type, comments):
```

Comment:
I would rename this to `create_and_add_unit` and make it a method.

Reply:
Good point. At some earlier point it did use some of the namespace, but it's now gone.